File: frontend/src/app/dto/task.dto.ts
import {ParentTaskDto} from "./parent-task.dto";
import {UserDto} from "./user.dto";
import {ProjectDto} from "./project.dto";

export class TaskDto {
  public taskId: string;
  public taskName: number;
  public startDate: string;
  public endDate: string;
  public priority: number;
  public status: string;
  public isParentTask: boolean
  public userId: number;
  public projectId: number;
  public parentTaskId: number;
  public userDto: UserDto = new UserDto();
  public parentTaskDto: ParentTaskDto = new ParentTaskDto();
  public projectDto: ProjectDto = new ProjectDto();
}

File: backend/src/main/java/com/pm/dto/TaskDto.java
package com.pm.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class TaskDto {
    private long taskId;
    private String taskName;
    private LocalDate startDate;
    private LocalDate endDate;
    private int priority;
    private String status;
    private boolean isParentTask;
    private long userId;
    private long projectId;
    private long parentTaskId;
    private ParentTaskDto parentTaskDto = new ParentTaskDto();
    private UserDto userDto = new UserDto();
    private ProjectDto projectDto = new ProjectDto();
}
